recipe/build.sh

```diff
@@ -3,6 +3,10 @@
 set -e
 set -x
 
+if [ "$(uname)" = "Linux" ] ; then
+  export LDFLAGS="${LDFLAGS} -Wl,-rpath-link,${PREFIX}/lib"
```

Comment:
This looks unexpected. Isn't `conda-build` supposed to take care of that?

Reply:
Agree, but otherwise `ld` fails. The strange thing is within conda-build only `-Wl,-rpath,$PREFIX/lib` is passed:
```
FAILED: release/plasma_store_server
: && $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++  -Wno-noexcept-type -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I$PREFIX
/include -fdebug-prefix-map==/usr/local/src/conda/- -fdebug-prefix-map==/usr/local/src/conda-prefix -fdiagnostics-color=always -O3 -DNDEBUG  -Wall -msse4.2  -fPIC -O3 -DNDEBUG  -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z
,now -Wl,--disable-new-dtags -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib   -rdynamic src/plasma/CMakeFiles/plasma_store_server.dir/external_store.cc.o src/plasma/CMakeFiles/plasma_store_server.dir/hash_table_store.cc.o src/plasma/CMakeFiles/plas
ma_store_server.dir/dlmalloc.cc.o src/plasma/CMakeFiles/plasma_store_server.dir/events.cc.o src/plasma/CMakeFiles/plasma_store_server.dir/eviction_policy.cc.o src/plasma/CMakeFiles/plasma_store_server.dir/plasma_allocator.cc.o src/plasma/
CMakeFiles/plasma_store_server.dir/store.cc.o src/plasma/CMakeFiles/plasma_store_server.dir/thirdparty/ae/ae.c.o  -o release/plasma_store_server  -Wl,-rpath,/tmp/build/release: release/libplasma.so.13.0.0 release/libarrow.so.13.0.0 $PREFI
X/lib/libdouble-conversion.a $PREFIX/lib/libbrotlienc.so $PREFIX/lib/libbrotlidec.so $PREFIX/lib/libbrotlicommon.so $PREFIX/lib/libglog.so -ldl -pthread -lrt
```

However compiling arrow with the same flags within a conda environment using the [conda compilers](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#compiler-packages) (gcc_linux-64, gxx_linux-64, gfortran_linux-64) both `-Wl,-rpath,$CONDA_PREFIX/lib` and `-Wl,-rpath-link,$CONDA_PREFIX/lib` are passed.

Other feedstocks are passing rpath flags explicitly too:
- [-rpath](https://github.com/search?q=org%3Aconda-forge+-rpath&type=Code)
- [-rpath-link](https://github.com/search?q=org%3Aconda-forge+-rpath-link&type=Code)

Reply:
@jjhelmus Do you know if this is regular practice?

Reply:
Adding `-Wl,-rpath-link...` is common and is a acceptable addition.  `-rpath` alone used to be sufficient but `ld` changed the behavior of the flag with respect to sysroots and therefore `-rpath-link` is now needed in certain cases.  This should be included automatically in LDFLAGS if an activation package is used which included AnacondaRecipes/aggregate#103

Reply:
@pitrou may We merge it then?

Reply:
Yes. Do you have merge rights?

Reply:
@pitrou You're a recipe maintainer as well, so You should be able to merge.

Reply:
Ok. I preferred to let you be the final judge since you're the PR author.